### PR TITLE
Added ERL computation in Circuit through SPISIM launched in non-graphical mode

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: pyaedt/circuit.py [hunk 1]
```diff
@@ -1684,3 +1687,150 @@ def import_edb_in_circuit(self, edb_path):
...
+            Permitted reflection (Rho). See SPISIM Help for info. Default is ``0.18``.
+        reflections_lenght : float, optional
+            Length of the reflections: how many UI will be used to calculate ERL. Default is ``1000``.
+        modulation_type : str, optional
+           Modulations type: signal modulation type "``NRZ``" or "``PAM4``". Default is "``NRZ``".
+
+        Returns
+        -------
+
+        """
```
Comment (SMoraisAnsys) [inline pyaedt/circuit.py:1752]:
```suggestion
        Returns
        -------
        bool or dict
            A dictionary with the result from the spisimExe command, ``False`` when failed.
        """
```

File: pyaedt/circuit.py [hunk 2]
```diff
@@ -1684,3 +1687,150 @@ def import_edb_in_circuit(self, edb_path):
...
+            fp.write("NCYCLES = {}\n".format(reflections_lenght))
+
+        cfgCmmd = '-i %s -v CFGFILE="%s"' % (snp_file, cfg_file)
+        command = [spisimExe, "CalcERL", cfgCmmd]
+        # Debug('%s %s' % (cmdList[0], ' '.join(arguments)))
+        # try up to three times to be sure
+        out_processing = os.path.join(self.working_directory, "spsim_erl_out.txt")
+        my_env = os.environ.copy()
+        for env, val in settings.aedt_environment_variables.items():
+            my_env[env] = val
```
Comment (SMoraisAnsys) [inline pyaedt/circuit.py:1817]:
```suggestion
        my_env.update(settings.aedt_environment_variables)
```

File: pyaedt/circuit.py [hunk 3]
```diff
@@ -1684,3 +1687,150 @@ def import_edb_in_circuit(self, edb_path):
...
+           Modulations type: signal modulation type "``NRZ``" or "``PAM4``". Default is "``NRZ``".
+
+        Returns
+        -------
+
+        """
+        if is_linux:
+            spisimExe = os.path.join(self.desktop_install_dir, "spisim/SPISim/modules/ext/SPISimJNI_LX64.exe")
+        else:
+            spisimExe = os.path.join(self.desktop_install_dir, r"spisim\SPISim\modules\ext\SPISimJNI_WIN64.exe")
```
Comment (SMoraisAnsys) [inline pyaedt/circuit.py:1756]:
```suggestion
        exec_name = "SPISimJNI_LX64.exe" if is_linux else "SPISimJNI_WIN64.exe"
        spisimExe = os.path.join(self.desktop_install_dir, "spisim", "SPISim", "modules", "ext", exec_name)
```

**Review by SMoraisAnsys — APPROVED** [review]

**Review by ring630 — APPROVED** [review]
LGTM

**Comment by SMoraisAnsys** [comment]
@maxcapodi78 I thought that one of my changes was the reason of the error but I'm not able to reproduce it locally when executing the added test alone. I'm relaunching the solver tests

**Review by SMoraisAnsys — APPROVED** [review]

**Pull request merged**